pkg/tracing/tracing.go

```diff
@@ -72,6 +73,16 @@ func StartSpan(ctx context.Context, operationName string, opts ...opentracing.St
 	return span, opentracing.ContextWithSpan(ctx, span)
 }
 
+// DoInSpanWtihErr executes function doFn inside new span with `operationName` name and hooking as child to a span found within given context if any.
+// It uses opentracing.Tracer propagated in context. If no found, it uses noop tracer notification.
+// It logs the error inside the new span created, which differentiates it from DoInSpan and DoWithSpan.
+func DoInSpanWithErr(ctx context.Context, operationName string, doFn func(context.Context) error, opts ...opentracing.StartSpanOption) {
+	span, newCtx := StartSpan(ctx, operationName, opts...)
+	defer span.Finish()
+	err := doFn(newCtx)
+	span.LogFields(opentracing_log.Error(err))
```

Comment:
1) I'm not sure about the behavior if `err` is `nil` (whether it still logs error), but maybe we could check `err != nil` before we log error?
2) I think we can replace this with method from the `ext` package, see https://pkg.go.dev/github.com/opentracing/opentracing-go@v1.2.0/ext#LogError. This will set all necessary things on the span for us

Reply:
Good point @matej-g. Thanks!
I've made the changes in a recent commit. Let me know if they're okay.